### DISCUSSION TITLE
Inverter Limit goes above 100%

### DISCUSSION THREAD

**Discussion by Neffez** [tbnobody/OpenDTU#2399] [Q&A]
Hi, I'm able to set my Inverter, which should be able to do 2000W, to 2250W. Is this normal or a bug? Is it some kind of overclocking or does it actually not work and I will get 2000W max. anyways? I did not find any info about this anywhere else. Thanks!

![image](https://github.com/user-attachments/assets/3380820c-5169-46e8-b64f-2e14d6b21e81)

**Comment by stefan123t** [comment]
@Neffez please provide the inverter details information with Hardware part number, etc and grid profile hex dump as this may be related to your inverter firmware veraion.

**Comment by Neffez** [comment]
```
Production Year	2024
Production Week	0
Model	HMS-2000-4T
Detected max. Power	2,000 W
Bootloader Version	0.1.1
Firmware Version	1.0.27
Firmware Build Date	2023-06-05 10:24:00
Hardware Part Number	270692628
Hardware Version	01.10
```

`0a 00 20 01 00 0c 08 fc 07 a3 00 0f 09 e2 00 1e 06 4a 00 14 0a 55 00 14 0a c8 00 0a 09 e2 10 03 13 88 12 c0 00 14 13 ec 00 14 12 8e 00 32 14 50 00 32 20 00 00 01 30 03 02 58 09 e2 07 a3 13 9c 13 56 40 00 07 d0 00 10 50 01 00 01 13 9c 01 90 00 10 00 00 60 00 00 01 09 e2 0a 5a 02 15 80 01 00 00 08 5b 01 2c 08 b7 09 41 09 9d 01 2c 00 64 90 00 00 00 00 5f b0 00 00 00 01 f4 00 5f 70 02 00 01 27 10 a0 02 00 00 00 00 d8 8b`

Is that what you need?

**Reply by stefan123t** [reply]
Can you also reply with some logs where you set the ActivePowerLimit to 100%/2000W. Make sure to include the SystemConfigPara Command which reads back the relevant Setting from the inverter.

**Reply by Neffez** [reply]
```
11:14:47.459 > TX ActivePowerControl 865.00 MHz --> 51 A0 0C B5 80 80 16 48 24 81 0B 00 03 E8 00 01 D0 40 C2 
11:14:47.503 > Interrupt received
11:14:47.549 > RX 865.00 MHz --> D1 A0 0C B5 80 80 16 48 24 81 00 00 0B 00 14 07 2B | -75 dBm
11:14:49.442 > RX Period End
11:14:49.442 > Success
11:14:50.934 > Fetch inverter: 1164XXXXXXXX
11:14:50.982 > TX RealTimeRunData 865.00 MHz --> 15 A0 0C B5 80 80 16 48 24 80 0B 00 67 31 D9 19 00 00 00 00 00 00 00 00 6B 5C 5C 
11:14:51.027 > Interrupt received
11:14:51.074 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 01 00 01 01 99 01 99 00 4F 00 4C 01 47 01 3A 00 00 88 | -75 dBm
11:14:51.120 > Interrupt received
11:14:51.168 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 02 09 E5 00 00 09 B5 00 30 00 2E 01 9A 01 9C 00 4D F1 | -75 dBm
11:14:51.211 > Interrupt received
11:14:51.256 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 03 00 50 01 3D 01 4D 00 00 09 83 00 00 09 C6 00 2F BF | -75 dBm
11:14:51.302 > Interrupt received
11:14:51.346 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 04 00 31 08 EC 13 87 04 CC 00 01 00 35 03 E7 00 5A F1 | -75 dBm
11:14:51.392 > Interrupt received
11:14:51.438 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 85 00 02 3F C2 8C | -75 dBm
11:14:51.483 > RX Period End
11:14:51.483 > Success
```
I can't find SystemConfigPara in the console, what do I have to do?

**Reply by stefan123t** [reply]
@Neffez it is my understanding that after an ActivePowerControl command the DTU will query your inverters ActivePowerLimit setting using the SystemConfigPara Command.
https://github.com/tbnobody/OpenDTU/blob/3dc70ab40aade8b7eb9ed9a1c6605ca326299d18/lib/Hoymiles/src/commands/SystemConfigParaCommand.cpp#L7-L19
The **Limit in Percent** is then parsed in the `SystemConfigParaParser.cpp`:
https://github.com/tbnobody/OpenDTU/blob/3dc70ab40aade8b7eb9ed9a1c6605ca326299d18/lib/Hoymiles/src/parser/SystemConfigParaParser.cpp#L7-L17

I do not know why this does not take place in your log snippet, but I believe that it may take several minutes before this is queried from the Inverter. Maybe you could analyse a longer Logfile and search for SystemConfigPara ?

**Reply by stefan123t** [reply]
Regarding your Grid Profile I had a peek into our list of Grid profiles and could find it with the checksum (last two bytes) `d8 8b` twice in the list, so this appears to be a genuine original Grid Profile XX - EN 50549-1:2019:

Country | Model | Grid Profile (from OpenDTU live data page) | Grid Profile Name (opt) | Inverter Firmware Version | Inverter Firmware | Creation Date
--- | --- | --- | --- | --- | --- | ---
PL | HMS-2000-4T | 0a 00 20 01 00 0c 08 fc 07 a3 00 0f 09 e2 00 1e 06 4a 00 14 0a 55 00 14 0a c8 00 0a 09 e2 10 03 13 88 12 c0 00 14 13 ec 00 14 12 8e 00 32 14 50 00 32 20 00 00 01 30 03 02 58 09 e2 07 a3 13 9c 13 56 40 00 07 d0 00 10 50 01 00 01 13 9c 01 90 00 10 00 00 60 00 00 01 09 e2 0a 5a 02 15 80 01 00 00 08 5b 01 2c 08 b7 09 41 09 9d 01 2c 00 64 90 00 00 00 00 5f b0 00 00 00 01 f4 00 5f 70 02 00 01 27 10 a0 02 00 00 00 00 d8 8b | XX - EN 50549-1:2019 | 2.0.1 | 1.0.27 | 2023-06-05 10:24:00
FR | HMS-2000-4T | 0a 00 20 01 00 0c 08 fc 07 a3 00 0f 09 e2 00 1e 06 4a 00 14 0a 55 00 14 0a c8 00 0a 09 e2 10 03 13 88 12 c0 00 14 13 ec 00 14 12 8e 00 32 14 50 00 32 20 00 00 01 30 03 02 58 09 e2 07 a3 13 9c 13 56 40 00 07 d0 00 10 50 01 00 01 13 9c 01 90 00 10 00 00 60 00 00 01 09 e2 0a 5a 02 15 80 01 00 00 08 5b 01 2c 08 b7 09 41 09 9d 01 2c 00 64 90 00 00 00 00 5f b0 00 00 00 01 f4 00 5f 70 02 00 01 27 10 a0 02 00 00 00 00 d8 8b | XX - EN 50549-1:2019 | 2.0.1 | 1.0.27 | 2023-06-05 10:24:00

**Reply by Neffez** [reply]
I've got one after leaving the console open for a while

```
10:16:49.238 > Fetch inverter: 1164XXXXXXXX
10:16:49.238 > Request SystemConfigPara
10:16:49.283 > TX RealTimeRunData 865.00 MHz --> 15 A0 0C B5 80 80 16 48 24 80 0B 00 67 33 1D 01 00 00 00 00 00 00 00 00 3A B9 36 
10:16:49.329 > Interrupt received
10:16:49.373 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 01 00 01 01 92 01 93 00 46 00 43 01 1C 01 12 00 00 FC | -74 dBm
10:16:49.418 > Interrupt received
10:16:49.465 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 02 0A DB 00 00 0A A3 00 2D 00 2B 01 99 01 96 00 43 C6 | -74 dBm
10:16:49.523 > Interrupt received
10:16:49.572 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 03 00 47 01 12 01 21 00 00 0A 72 00 00 0A C2 00 2B 1A | -74 dBm
10:16:49.617 > Interrupt received
10:16:49.663 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 04 00 2E 09 10 13 8A 04 2B 00 00 00 2D 03 E7 00 58 E2 | -74 dBm
10:16:49.709 > Interrupt received
10:16:49.755 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 85 00 01 F9 75 FE | -74 dBm
10:16:49.802 > RX Period End
10:16:49.802 > Success
10:16:49.849 > TX SystemConfigPara 865.00 MHz --> 15 A0 0C B5 80 80 16 48 24 80 05 00 67 33 1D 01 00 00 00 00 00 00 00 00 F4 B6 F9 
10:16:49.895 > Interrupt received
10:16:49.941 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 81 00 01 03 E8 00 00 03 E8 00 00 00 00 00 00 3C F8 B2 | -74 dBm
10:16:50.030 > RX Period End
10:16:50.030 > Success
```

**Reply by stefan123t** [reply]
@Neffez I can not see a problem here:

SystemConfigPara
```
15 A0 0C B5 80 80 16 48 24 80 05 00 67 33 1D 01 00 00 00 00 00 00 00 00 F4 B6 F9 
95 A0 0C B5 80 80 16 48 24 81 00 01 03 E8 00 00 03 E8 00 00 00 00 00 00 3C F8 B2
^^ ^^^^^^^^^^^ ^^^^^^^^^^^ ^^ ^^^^^ ^^^^^ ^^^^^ ^^^^^ ^^^^^ ^^^^^ ^^^^^ ^^^^^ ^^ 
ID Source Addr Target Addr Idx  ?   Limit   ?     ?     ?     ?     ?   CRC16 CRC8 
```

`0x03E8` = `0d1000` so this should yield 100.0 % in the UI too.

Are you sure you have waited long enough for the UI to show the update from the SystemConfigPara Response after changing the ActivePowerLimit ?

**Reply by Neffez** [reply]
Yes it showed 100% in the UI as well. So did you actually want me to set it to 112,5%? I can set it to 112,5% tomorrow and post the results.

**Reply by Neffez** [reply]
```
11:03:04.599 > TX SystemConfigPara 865.00 MHz --> 15 A0 0C B5 80 80 16 48 24 80 05 00 67 35 CA D7 00 00 00 00 00 00 00 00 6A AB 7D 
11:03:04.653 > Interrupt received
11:03:04.700 > RX 865.00 MHz --> 95 A0 0C B5 80 80 16 48 24 81 00 01 04 65 00 00 03 E8 00 00 00 00 00 00 99 9F FA | -77 dBm
11:03:04.803 > RX Period End
11:03:04.803 > Success
```

Here you go, it shows `04 65` which is 112.5%.

**Reply by stefan123t** [reply]
You are right `0x04 65` = `0d1125` or 112,5%.
Okay, so OpenDTU is just telling you what it receives from the Inverter. 

Is this Inverter somehow modified e.g. by something  like the latest Hoymiles Firmware Update or the Flole-Mod ?
I am out of answers or at least I do not have further questions from my side :slightly_smiling_face:

**Reply by Neffez** [reply]
It's a new inverter I've bought from an ordinary online shop, so I don't think it's modified. I'm just curious what it means that it can go above 100%. It seems like there is nobody else with an inverter like this, which is interesting. Thanks anyways.
